### PR TITLE
Added new media types

### PR DESCRIPTION
**Pull request by prbinu**
IANA Consideration section will added once we have new media type assignment details available

### CODE REVIEW

**Comment by prbinu** [comment]
@danmarg @abrotman 
Please review and merge

File: reporting.md [hunk 1]
```diff
@@ -355,14 +435,16 @@ compute cost.
 
 ## Email Transport
 
-The report MAY be delivered by email. No specific MIME message structure is
-required. It is presumed that the aggregate reporting address will be equipped
+The report MAY be delivered by email. To make the reports machine parsable
+for the receivers, we define a top-level media type `multipart/report` with
+a new parameter `report-type=tlsrpt`. Inside it, there are two parts: The
+first part is human readable, typically `text/plain`, and the second part is
+machine readable with a new media type defined called `application/tlsrpt+gzip`.
```
Comment (abrotman) [inline reporting.md]: Shouldn't this first one be "application/tlsrpt+json"?

File: reporting.md [hunk 2]
```diff
@@ -389,14 +471,49 @@ gzip` if compressed (see [@!RFC6713]), and `application/json` otherwise.
...
+### tlsrpt Example 
+```
+ From: tlsrpt@mail.sender.example.com
+     Date: Fri, May 09 2017 16:54:30 -0800
+     To: mts-sts-tlsrpt@example.net
+     Subject: Report Domain: example.net
+         Submitter: mail.sender.example.com
+         Report-ID: <735ff.e317+bf22029@example.net>
+     MIME-Version: 1.0
+     Content-Type: multipart/report; report-type=tlsrpt
```
Comment (abrotman) [inline reporting.md]: Quote the Content-Type bits?  Needs a trailing ";"

File: reporting.md [hunk 3]
```diff
@@ -481,7 +481,7 @@ the rest.
          Submitter: mail.sender.example.com
          Report-ID: <735ff.e317+bf22029@example.net>
      MIME-Version: 1.0
-     Content-Type: multipart/report; report-type=tlsrpt
+     Content-Type: multipart/report; report-type="tlsrpt"
```
Comment (abrotman) [inline reporting.md]: Per Alexey, needs trailing ";"

**Comment by abrotman** [comment]
Added some comments, wasn't sure if you wanted me to correct those or you'd prefer.

File: reporting.md (hunk 3 above)
Comment (prbinu) [inline reporting.md]: fixed

**Comment by prbinu** [comment]
> Added some comments, wasn't sure if you wanted me to correct those or you'd prefer.
@abrotman, Sure, but I would suggest to merge this PR (if you are ok with the change) and raise another PR with your changes. One main task left is to update IANA consideration section.

**Comment by prbinu** [comment]
@danmarg @abrotman 
Changes complete. Please review and merge

**Pull request merged**